centaurCwlRunner/src/main/scala/centaur/cwl/OutputManipulator.scala

```diff
@@ -85,39 +101,39 @@ object OutputManipulator extends Poly1 {
         * 2) the size calculated from the value of the "contents" field of the metadata
         * 3) the size calculated from the file itself
         */
-      val defaultSize = sizeContent.orElse(sizeFile).map(Json.fromLong).getOrElse(Json.Null)
+      val defaultSize = sizeContent.orElse(sizeFile(path)).map(Json.fromLong).getOrElse(Json.Null)
       val size = valueOrNull("size", defaultSize)
 
+      val basename: Option[Json] = if (isInsideDirectory) {
+        Option(valueOrNull("basename", path.exists.option(path.nameWithoutExtension).map(Json.fromString).getOrElse(Json.Null)))
+      } else None
```

Comment:
I'm not getting the "inside directory" logic here

Reply:
Yeah I meant to add a comment and forgot. Apparently the conformance tests expect the files listed in the `listing` field for directories to have `basename` filled in, but only there and not for other top level files...
So this flag is there to decide if we should add basename